src/compiler/scala/tools/nsc/backend/jvm/opt/BytecodeUtils.scala

```diff
@@ -123,6 +123,8 @@ object BytecodeUtils {
 
   def isReference(t: Type) = t.getSort == Type.OBJECT || t.getSort == Type.ARRAY
 
+  def isScala(classNode: ClassNode) = classNode.attrs != null && classNode.attrs.asScala.exists(a => a.`type` == BTypes.ScalaAttributeName || a.`type` == BTypes.ScalaSigAttributeName)
```

Comment:
i just was a bit confused why this works, because the scala pickle is stored in the `ScalaSignature` *annotation* nowadays (no longer in the `ScalaSig` *attribute*), but it turns out we still put the attribute. do you know if we have a test for that? https://github.com/scala/scala/blob/2e40aa5871859b9ab6bdc03d56a81746fa90b1e5/src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala#L1119-L1121